pkg/kubectl/cmd/util/factory.go

```diff
@@ -333,11 +336,15 @@ func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
 	}
 
 	mapper := registered.RESTMapper()
-	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
-	if err == nil {
+
+	var cachedDiscoveryClient clientdiscovery.DiscoveryInterface
+	if discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg); err == nil {
+		cacheDir := computeDiscoverCacheDir(filepath.Join(homedir.HomeDir(), ".kube"), cfg.Host)
+		cachedDiscoveryClient = NewCachedDiscoveryClient(discoveryClient, cacheDir, time.Duration(10*time.Minute))
+
 		// register third party resources with the api machinery groups.  This probably should be done, but
 		// its consistent with old code, so we'll start with it.
-		if err := registerThirdPartyResources(discoveryClient); err != nil {
+		if err := registerThirdPartyResources(cachedDiscoveryClient); err != nil {
```

Comment:
Question: does it make sense to cache third party resources? They might change any time. /cc @deads2k